torch/distributions/kl.py

```diff
@@ -278,6 +279,31 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    if p.batch_shape != q.batch_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different batch shapes cannot be computed")
+
+    def _batch_trace(bmat):
+        mat_size = bmat.size(-1)
+        values = torch.stack([torch.trace(M) for M in bmat.contiguous().view((-1, mat_size, mat_size))])
+        return values.view(bmat.shape[:-2])
+
+    def _batch_mm(bmat1, bmat2):
+        mat_size = bmat1.size(-1)
+        bmat1_squash = bmat1.contiguous().view((-1, mat_size, mat_size))
+        bmat2_squash = bmat2.contiguous().view((-1, mat_size, mat_size))
+        return bmat1_squash.bmm(bmat2_squash).view_as(bmat1)
+
+    term1 = _batch_diag(q.scale_tril).abs().log().sum(-1) - _batch_diag(p.scale_tril).abs().log().sum(-1)
```

Comment:
I believe our `.scale_tril` is required to already have positive diagonal, so the `.abs()` should be unnecessary.

Reply:
The trace of `.scale_tril` is positive, which might not ensure that the individual elements are positive. This is why I added it.

Reply:
The individual elements are guaranteed to be positive. See https://github.com/pytorch/pytorch/blob/03a6952/torch/distributions/transforms.py#L504

Reply:
Oh, I didn't see that. Sorry.